src/vanilla/store.ts

```diff
@@ -520,10 +518,10 @@ export const createStore = () => {
           visit(m)
         }
       }
-      // The algorithm calls for pushing onto the front of the list. Since
-      // Array.unshift is likely slower than pushing onto the end, we simply
-      // push and will have to iterate in reverse order later.
-      topsortedAtoms.push(n)
+      // The algorithm calls for pushing onto the front of the list. For
+      // simplicity, we simply append items in order, and will iterate
+      // in reverse order later.
```

Comment:
It's no longer in reverse order, right?
Oh, wait. That was my misunderstanding. So, your suggestion was correct. My bad.
I meant to fill it from the end...

Reply:
No worries, I just updated it